### ISSUE TITLE
Databento MBO: F (Fill) records decoded as book deltas corrupt the L3 book (phantom orders, crossed book)

### ISSUE THREAD

**Issue by taozle** [nautechsystems/nautilus_trader#4445]
## Bug report

- **Affected**: `DatabentoDataLoader.from_dbn_file()` (and any path through `decode_mbo_msg`) with GLBX MBO data
- **Version**: 1.230.0 (present on current `develop` as well)
- **OS / Python**: Linux x86_64, Python 3.13

## Summary

`decode_mbo_msg` converts Databento MBO records with `action='F'` (Fill) into
`OrderBookDelta` with `BookAction::Update` (`parse_book_action` maps
`'F' => Update`). Per CME MDP3 semantics as normalized by Databento, **Fill
records never affect the book**: they are attribution records for the resting
order, and the actual book impact arrives as the *explicit* Cancel/Modify
records of the same match event. See Databento's docs on MBO actions —
Trade/Fill records carry no book update; book maintenance is driven by
A/C/M/R only.

The `dbn` crate itself (which this adapter depends on) documents the
[`Action`](https://docs.rs/dbn/latest/dbn/enums/enum.Action.html) variants as:

> **Fill** — "An existing order was filled. **Does not affect the book.**"
> **Trade** — "An aggressing order traded. Does not affect the book."
> **None** — "Has no effect on the book, but may carry `flags` or other information."

Decoding `F` as a delta corrupts the book in two ways:

1. **Displayed orders**: the `F` record's `size` field is the *filled amount*,
   not the remaining size, so the absolute-size `Update` sets the wrong
   quantity (usually masked because the explicit `C`/`M` follows in the same
   event).
2. **Iceberg hidden-part fills (the lethal case)**: the hidden portion of an
   iceberg fills with an `F` whose `order_id` was never `Add`ed and is never
   `Cancel`ed. The `Update` delta materializes a **phantom order** that
   nothing ever deletes. These accumulate over a session.

## Observed impact

Replaying one full trading day of CME GLBX MBO data (NQ futures,
`DatabentoDataLoader.from_dbn_file(path, include_trades=True)` into an
`OrderBook(BookType::L3_MBO)`):

- By mid-session the book was **crossed by ~220 points** (best bid 29909.75 /
  best ask 29689.00) — the stale ask was a phantom order whose complete raw
  history is two records: `T`, `F` (no `A`, no `C`), i.e. an iceberg
  hidden-part fill.
- In a `BacktestEngine` (`book_type=L3_MBO`, `trade_execution=True`), market
  BUY orders then fill at the stale phantom ask, hundreds of points away from
  the prevailing market.
- Conversion accounting confirms the mechanism: the loader's delta count
  equals the file's `A+C+M+R+F` record count **exactly** (all 418,061 `F`
  records of that day became deltas).

## Reproduction

Any GLBX MBO day with iceberg activity reproduces it. Two checks that don't
need specific data:

```python
# 1) loader delta count == A+C+M+R+F (F records are being converted)
import databento as db
from nautilus_trader.adapters.databento.loaders import DatabentoDataLoader
from nautilus_trader.model.data import OrderBookDelta

path = "<any GLBX .mbo.dbn(.zst)>"
actions = {}
for rec in db.DBNStore.from_file(path):
    a = chr(rec.action) if hasattr(rec, "action") else None
    actions[a] = actions.get(a, 0) + 1

data = DatabentoDataLoader().from_dbn_file(path, include_trades=True)
n_deltas = sum(isinstance(d, OrderBookDelta) for d in data)
expected_book_actions = sum(actions.get(k, 0) for k in "ACMR")
print(n_deltas, expected_book_actions, expected_book_actions + actions.get("F", 0))
# n_deltas == A+C+M+R+F, not A+C+M+R

# 2) book integrity: replay deltas into OrderBook(L3_MBO) and assert
#    best_bid < best_ask periodically — fails once an iceberg hidden fill
#    has been converted.
```

Unit-level: construct a `dbn::MboMsg` with `action='F'` and an `order_id`
that was never added — `decode_mbo_msg` currently returns an `Update` delta
for it.

## Expected behavior

`F` records produce no delta and no trade (the trade is already conveyed by
the `T` record; converting `F` to trades would double-count volume).
`N` (None/status) records should likewise be skipped — currently they hit the
`anyhow::bail!` arm and error the decode.

## Proposed fix

In `decode_mbo_msg`, filter `'F'`/`'N'` before `parse_book_action` and return
`(None, None)`; drop the `'F' => Update` arm from the Databento
`parse_book_action`. I have a patch with tests ready and am happy to open a
PR.

**Comment by cjdsellers** [comment]
Hi @taozle,

Thanks for the clear report and patch. The core decoder change is correct, and the 188 decoder tests pass locally.

There is one live-path gap before this can merge. Databento documents that `N` records may carry `F_LAST`, but `decode_mbo_msg` now returns `(None, None)` and the live MBO caller immediately continues unless it receives `Data::Delta`. A buffered delta followed by `N | F_LAST` therefore remains stranded or merges with the next event.

Please make the live MBO path process raw event-boundary flags independently from the optional decoded payload, and add a regression test covering a non-terminal delta followed by `N | F_LAST`.

**Comment by taozle** [comment]
Thanks for the catch! I've updated the PR. PTAL.
